pkg/pillar/cmd/volumemgr/create.go

```diff
@@ -22,46 +22,60 @@ func createVolume(ctx *volumemgrContext, status types.VolumeStatus) (bool, strin
 		log.Infof("createVolume(%s) from container %s", status.Key(), status.ReferenceName)
 		return createContainerVolume(ctx, status, status.ReferenceName)
 	}
-	log.Infof("createVolume(%s) from disk %s", status.Key(), status.FileLocation)
-	return createVdiskVolume(ctx, status, status.FileLocation)
+	log.Infof("createVolume(%s) from disk %s", status.Key(), status.ReferenceName)
+	return createVdiskVolume(ctx, status, status.ReferenceName)
 }
 
 // createVdiskVolume does not update status but returns
 // new values for VolumeCreated, FileLocation, and error
```

Comment:
Is the second return value still a path name which should be used by the called to update FileLocation even in the error cases?

Reply:
The returned is the path to the root disk. The argument now is the image reference name, e.g. `lf-edge/ubuntu:19.04`m which contains a VM disk, the same way that it is for a container, e.g.`library/nginx:latest`

The return for a container is the path to the root directory; for a vdisk, it is the path to the root disk, i.e. inside that directory.

Reply:
On completion you return a path, but on failure or 'no work needed' you return the input. I question whether that is still useful. I think the caller uses that to update FileLocation.

Reply:
Ah, you are referring to the various failure lines like this?

```go
		return created, ref, errors.New(errStr)
```

I didn't understand that you were referring to that. I think the assumption was that if an error is returned, the string value would be ignored. But it makes sense to make it empty in that case.

Will update.

Reply:
Updated.

Reply:
Do the caller(s) ignore it on error?

Reply:
Not precisely. See below (truncated):

```go
	volumeCreated, fileLocation, err = createVolume(ctx, d.status)
	d.VolumeCreated = volumeCreated
	d.FileLocation = fileLocation
	result := worker.WorkResult{
		Key:         w.Key,
		Description: d,
	}
	if err != nil {
		result.Error = err
		result.ErrorTime = time.Now()
	}
```

It sets the file location, but it also sets the error.

It took a while to trace it through, but eventually this gets passed on to the work result, which gets consumed [here](https://github.com/lf-edge/eve/blob/a324e6fbfe766a8a50ba81d03fb1f0ba30c6d837/pkg/pillar/cmd/volumemgr/updatestatus.go#L380-L403).

It _does_ set the `VolumeStatus.FileLocation`, but it also sets the error and the `VolumeCreated`, so it should be safe.

Reply:
It would be better to not have the input key show up in FileLocation. So you need to change the above caller code IMHO

Reply:
@eriknordmark I don't understand. The return calls from `createVdiskVolume` now return either `""` (in the case of an error) or the actual root disk file location. What more do we need to change?

Reply:
That makes the callers set FileLocation to "" as opposed to keeping it unchanged. Having the callers not update FileLocation on error seems to make more sense with your changes.

Reply:
Changed it to set `FileLocation` only if `err == nil`.